src/loadRc.js

```diff
@@ -32,26 +32,44 @@ module.exports = function loadRc(
   }
 
   function parseExtensionlessRcFile(content: ?string): ?cosmiconfig$Result {
-    if (!content) return null;
-    const pasedConfig = options.rcStrictJson
+    if (content == null) return null;
+
+    const isEmpty = content === '';
+    if (isEmpty) {
+      return {
+        config: undefined,
+        filepath,
+        isEmpty,
+      };
+    }
+
+    const parsedConfig = options.rcStrictJson
       ? parseJson(content, filepath)
       : yaml.safeLoad(content, { filename: filepath });
     return {
-      config: pasedConfig,
+      config: parsedConfig,
       filepath,
     };
   }
 
   function loadRcWithExtensions() {
-    let foundConfig = null;
+    let foundConfig: ?cosmiconfig$Result = null;
```

Comment:
I had wanted to refactor this some time back to avoid the code repetition. @olsonpm, @davidtheclark, what do you think of this implementation for `loadRcWithExtensions`:

```js
function loadRcWithExtensions() {
  function loadExtn(
    extn: string,
    parse: (content: string, filename: string) => Object
  ) {
    // Check the result from the previous `loadExtn` invocation. If result
    // isn't null, just return that.
    return result => {
      if (result != null) return result;

      // Try to load the rc file for the given extension.
      return funcRunner(readRcFile(extn), [
        content => {
          if (content == null) return;

          const fp = `${filepath}.${extn}`;
          return content === ''
            ? { config: undefined, filepath: fp, isEmpty: true }
            : { config: parse(content, fp), filepath: fp };
        },
      ]);
    };
  }

  const parseYml = (content: string, filename: string) =>
    yaml.safeLoad(content, { filename });

  return funcRunner(!options.sync ? Promise.resolve() : undefined, [
    loadExtn('json', parseJson),
    loadExtn('yaml', parseYml),
    loadExtn('yml', parseYml),
    loadExtn('js', requireFromString),
  ]);
}
```

Reply:
I think I remember reading this conversation you guys had.  like you, I tend to go overboard on removing code repetition so davidtheclark would be better suited to say one way or the other.  I agree it's good to revisit with the amount of repeated code I added.

Reply:
Looks good. My only suggestion is spelling out `extension` instead of abbreviating it 😄

Reply:
That's side effects of [working on Go :wink:](https://github.com/golang/go/wiki/CodeReviewComments#variable-names)

Reply:
pushed